### PR TITLE
Update API doc to use the field name in description

### PR DESCRIPTION
**Pull request by ravi-nal**
#### What type of PR is this?
/kind api-change
/kind documentation

#### What this PR does / why we need it:
This PR focuses on updating the documentation, for the following issue, to use the field name in description. No other documentation changes. The documentation is updated for the API groups - discovery and storage.

https://github.com/kubernetes/kubernetes/issues/113438

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ravi-nal** [comment]
@xing-yang @wojtek-t @liggitt Please review when you get a chance.

File: staging/src/k8s.io/api/discovery/v1beta1/types.go [hunk 1]
```diff
@@ -168,15 +168,15 @@ type EndpointPort struct {
 	// * must start and end with an alphanumeric character.
 	// Default is empty string.
 	Name *string `json:"name,omitempty" protobuf:"bytes,1,name=name"`
```
Comment (liggitt) [inline staging/src/k8s.io/api/discovery/v1beta1/types.go:170]: ^ change "The name of this port" → "name represents the name of this port."

File: staging/src/k8s.io/api/discovery/v1beta1/types.go [hunk 2]
```diff
@@ -168,15 +168,15 @@ type EndpointPort struct {
 	// * must start and end with an alphanumeric character.
```
Comment (liggitt) [inline staging/src/k8s.io/api/discovery/v1beta1/types.go:168]: the same changes are needed in staging/src/k8s.io/api/discovery/v1/types.go

File: staging/src/k8s.io/api/storage/v1/types.go [hunk 3]
```diff
@@ -38,16 +38,17 @@ type StorageClass struct {
...
+	// provisioner indicates the type of the provisioner.
 	Provisioner string `json:"provisioner" protobuf:"bytes,2,opt,name=provisioner"`
 
-	// Parameters holds the parameters for the provisioner that should
+	// parameters holds the parameters for the provisioner that should
 	// create volumes of this storage class.
 	// +optional
 	Parameters map[string]string `json:"parameters,omitempty" protobuf:"bytes,3,rep,name=parameters"`
 
 	// Dynamically provisioned PersistentVolumes of this storage class are
```
Comment (liggitt) [inline staging/src/k8s.io/api/storage/v1/types.go:49]: something like 
```suggestion
	// reclaimPolicy controls the reclaimPolicy for dynamically provisioned PersistentVolumes of this storage class
```

File: staging/src/k8s.io/api/storage/v1/types.go [hunk 4]
```diff
@@ -57,17 +58,17 @@ type StorageClass struct {
 	// +optional
 	MountOptions []string `json:"mountOptions,omitempty" protobuf:"bytes,5,opt,name=mountOptions"`
```
Comment (liggitt) [inline staging/src/k8s.io/api/storage/v1/types.go:59]: something like 
"Dynamically provisioned PersistentVolumes of this storage class are created with these mountOptions" → "mountOptions controls the mountOptions for dynamically provisioned PersistentVolumes of this storage class"

File: staging/src/k8s.io/api/storage/v1/types.go [hunk 5]
```diff
@@ -349,7 +350,7 @@ type CSIDriverSpec struct {
 	// +featureGate=CSIStorageCapacity
 	StorageCapacity *bool `json:"storageCapacity,omitempty" protobuf:"bytes,4,opt,name=storageCapacity"`
```
Comment (liggitt) [inline staging/src/k8s.io/api/storage/v1/types.go:351]: PodInfoOnMount and StorageCapacity field doc needs fixing

**Review by liggitt — COMMENTED** [review]
thanks for the PR, this is a good start. I noted a few fields that still needed updates, and storage/{v1alpha1,v1beta1}/types.go and discovery/v1/types.go need the same updates

**Comment by ravi-nal** [comment]
@liggitt Implemented all the suggestions. Pls. check again. Appreciate for the review.

File: staging/src/k8s.io/api/storage/v1/types.go [hunk 6]
```diff
@@ -310,29 +312,27 @@ type CSIDriverSpec struct {
...
+	// A driver has to be aware of this because it is only going to get a NodePublishVolume call for such a volume.
+	//
 	// For more information about implementing this mode, see
 	// https://kubernetes-csi.github.io/docs/ephemeral-local-volumes.html
-	// A driver can support one or more of these modes and
-	// more modes may be added in the future.
-	// This field is beta.
+	// A driver can support one or more of these modes and more modes may be added in the future.
 	//
+	// This field is beta.
```
Comment (liggitt) [inline staging/src/k8s.io/api/storage/v1/types.go:326]: @pohly is this field still beta?

**Review by liggitt — COMMENTED** [review]
one question to storage review about a pre-existing comment that just got reformatted in this PR

lgtm otherwise

/lgtm
/approve
/milestone v1.27

**Comment by ravi-nal** [comment]
> one question to storage review about a pre-existing comment that just got reformatted in this PR
> 
> lgtm otherwise
> 
> /lgtm /approve /milestone v1.27

Thanks @liggitt. I will wait for the response for your question from @pohly

**Pull request merged**